guacml/step_tree/model_runner.py

```diff
@@ -1,45 +1,73 @@
-from guacml.step_tree.hyper_param_optimizer import HyperParameterOptimizer
-from guacml.step_tree.model_result import ModelResult
-from .base_step import BaseStep
+import pandas as pd
+import numpy as np
 
 
-class ModelRunner(BaseStep):
-    def __init__(self, model, target, hyper_param_iterations, eval_metric):
+class ModelRunner():
+    def __init__(self, model, input, target, eval_metric, splitter):
         self.model = model
         self.target = target
-        self.hyper_param_iterations = hyper_param_iterations
         self.eval_metric = eval_metric
+        train_and_cv, holdout = splitter.split(input)
+        self.holdout = holdout.copy()
+        train, cv = splitter.split(train_and_cv)
+        self.train = train.copy()
+        self.cv = cv.copy()
+        self.features = None
+        self.hyper_params = None
 
-    def execute(self, input, metadata):
-        train_and_cv, holdout = self.splitter.split(input)
-        train, cv = self.splitter.split(train_and_cv)
-        features = self.model.select_features(metadata)
-        features = features[features != self.target]
-
-        hp_optimizer = HyperParameterOptimizer(self.model, train, cv, features,
-                                               self.target, self.eval_metric)
-        all_trials = hp_optimizer.optimize(self.hyper_param_iterations)
-        all_trials = all_trials.sort_values('cv error')
-        best = all_trials.iloc[0]
-
-        training_error, _ = self.score_model(train, features)
-        holdout_error, holdout_predictions = self.score_model(holdout, features)
-        holdout_row_errors = self.eval_metric.row_wise_error(holdout[self.target], holdout_predictions)
-
-        holdout = holdout.copy()
-        holdout['error'] = holdout_row_errors
-        holdout['prediction'] = holdout_predictions
-
-        return ModelResult(self.model,
-                           self.target,
-                           training_error,
-                           best['cv error'],
-                           holdout_error,
-                           holdout,
-                           metadata,
-                           best,
-                           all_trials)
-
-    def score_model(self, input, features):
-        predictions = self.model.predict(input[features])
-        return self.eval_metric.error(input[self.target], predictions), predictions
+    def train_model(self, features, hyper_params):
+        self.features = features
+        self.hyper_params = hyper_params
+        self.model.train(self.train[features], self.train[self.target], **hyper_params)
```

Comment:
This might be unrelated to this PR but I'm wondering why we don't also pass the eval metric to the model train step. Shouldn't we somehow tell the model that it should use our `eval_metric`? What am I missing?